nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -189,18 +195,20 @@ def make_dataset(self, data_dimension: tp.Optional[int], dataset: str) -> None:
 
             # Half the dataset for training.
             rng.shuffle(data[0].T)  # We randomly shuffle the columns.
-            self.X = data[0][::2]
-            self.y = data[1][::2]
-            self.num_data = len(self.X)
-            for cv in range(10):
-                train_range = np.arange(self.num_data) % 10 != cv
-                valid_range = np.arange(self.num_data) % 10 == cv
-                self.X_train += [self.X[train_range]]
-                self.y_train += [self.y[train_range]]
-                self.X_valid += [self.X[valid_range]]
-                self.y_valid += [self.y[valid_range]]
-            self.X_test = data[0][1::2]
-            self.y_test = data[1][1::2]
+            self.X = data[0]
+            self.y = data[1]
+            self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y, test_size=0.5, random_state=42)
```

Comment:
`random_state=42`: do we want it to be deterministic? using the randomstate linked to the parametrization would make the function stochastic while keeping the reproducibility property

Reply:
It was deterministic, so i let it.
If we change that, we have to change the tests.